packages/svelte/src/internal/client/dom/elements/misc.js

```diff
@@ -56,3 +57,17 @@ export function add_form_reset_listener() {
 		);
 	}
 }
+
+/**
+ * @param {string} text
+ */
+export function title(text) {
+	render_effect(() => {
+		const previous = document.title;
+		document.title = text;
+
+		return () => {
+			document.title = previous;
```

Comment:
This should handle another edge case - thoughts?
```suggestion
			// If something else changed the title in the meantime, keep it
			if (document.title === text) {
				document.title = previous;
			}
```

Reply:
Which case would this prevent? It actually seems to me like this might make bugs _more_ likely, since if a component had a `<title>` and also contained a component that set `document.title` in a user effect, it wouldn't revert either when the parent was removed

Reply:
This handles the case where `A` sets the title, then `B` is mounted and sets the title, then `A` updates the title (in other words it contains something like `<title>{foo}</title>`), then `B` is removed - in that case we wouldn't want the title to be reverted back to what `A` had initially, we want to preserve the title.

Setting `document.title` in a user effect manually feels a bit like "working against the framework", you should just do that declaratively.

They are both edge cases either way, so I'm ok with not adding this suggestion.